gbpservice/nfp/configurator/api/v1/controllers/controller.py

```diff
@@ -322,10 +327,13 @@ def pull_notifications(self):
             self.channel.close()
             return notifications
         except pika.exceptions.ConnectionClosed:
+            LOG.error("Caught ConnectionClosed exception."
+                      "Creating new connection")
             self.create_connection()
-            self.pull_notifications()
+            return notifications
         except pika.exceptions.ChannelClosed:
+            LOG.error("Caught ChannelClosed exception.")
             if msgs_acknowledged is False:
-                self.pull_notifications()
+                return self.pull_notifications()
```

Comment:
Should be attempt to do invoke this pull_notifications() again in exception handling? Shouldn't this be left for next call of pull_notifications?